churaverse-plugins-server/src/churarenPlugin/churarenBossAttackPlugin/churarenBossAttackPlugin.ts

```diff
@@ -0,0 +1,110 @@
+import { Player } from '@churaverse/player-plugin-server/domain/player'
+import { CHURAREN_CONSTANTS } from '@churaverse/churaren-core-plugin-server'
+import { BossAttackRequestEvent } from '@churaverse/churaren-boss-plugin-server/event/bossAttackRequestEvent'
+import '@churaverse/churaren-core-plugin-server/event/churarenStartTimerEvent'
+import '@churaverse/player-plugin-server/store/defPlayerPluginStore'
+import '@churaverse/churaren-boss-plugin-server/store/defBossPluginStore'
+import '@churaverse/network-plugin-server/store/defNetworkPluginStore'
+
+export class ChurarenBossAttackPlugin extends BaseGamePlugin {
+  public gameId = CHURAREN_CONSTANTS.GAME_ID
+  private bossAttackPluginStore!: BossAttackPluginStore
+  private mapPluginStore!: MapPluginStore
+
+  public listenEvent(): void {
+    super.listenEvent()
+    this.bus.subscribeEvent('init', this.init.bind(this))
+
+    const socketController = new SocketController(this.bus, this.store)
+    this.bus.subscribeEvent('registerMessage', socketController.registerMessage.bind(socketController))
+    this.bus.subscribeEvent(
+      'registerMessageListener',
+      socketController.setupMessageListenerRegister.bind(socketController)
+    )
+    this.bus.subscribeEvent('registerOnOverlap', this.registerOnOverlap.bind(this))
+  }
+
+  protected subscribeGameEvent(): void {
+    super.subscribeGameEvent()
+    this.bus.subscribeEvent('update', this.update)
+    this.bus.subscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.subscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected unsubscribeGameEvent(): void {
+    super.unsubscribeGameEvent()
+    this.bus.unsubscribeEvent('update', this.update)
+    this.bus.unsubscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.unsubscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected handleGameStart(): void {}
+
+  protected handleGameTermination(): void {
+    this.bossAttackPluginStore.bossAttacks.clear()
+  }
+
+  private readonly update = (ev: UpdateEvent): void => {
+    moveBossAttacks(ev.dt, this.bossAttackPluginStore.bossAttacks, this.mapPluginStore.mapManager.currentMap)
+    removeDieBossAttack(this.bossAttackPluginStore.bossAttacks, (bossAttackId: string) => {
+      const bossAttackHitMessage = new BossAttackHitMessage({ bossAttackId })
+      const networkPluginStore = this.store.of('networkPlugin')
+      networkPluginStore.messageSender.send(bossAttackHitMessage)
+    })
+  }
+
+  private init(): void {
+    initBossAttackPluginStore(this.store)
+    this.bossAttackPluginStore = this.store.of('bossAttackPlugin')
+    this.mapPluginStore = this.store.of('mapPlugin')
+  }
+
+  private readonly generateBossAttack = (ev: BossAttackRequestEvent): void => {
+    const bossPluginStore = this.store.of('bossPlugin')
+    const boss = bossPluginStore.bosses.get(ev.bossId)
```

Comment:
特に再利用性がない＆単純なので一つにまとめてしまっても良いかなと思います
```suggestion
    const boss = this.store.of('bossPlugin').bosses.get(ev.bossId)
```